sycl/test-e2e/syclcompat/atomic/atomic_arith.cpp

```diff
@@ -115,7 +132,35 @@ template <typename T> void test_atomic_ptr_arith() {
   syclcompat::free(init);
 }
 
+void test_atomic_arith_t1_t2() {
+  std::cout << __PRETTY_FUNCTION__ << std::endl;
+  using data_t = float;
+  using operand_t = int;
+
+  constexpr syclcompat::dim3 grid{4};
+  constexpr syclcompat::dim3 threads{32};
+  constexpr data_t sum = static_cast<data_t>(grid.x * threads.x);
+  constexpr data_t init = static_cast<data_t>(0);
+  constexpr operand_t operand = static_cast<operand_t>(1);
+
+  AtomicLauncher<atomic_fetch_add_kernel<data_t, operand_t>, data_t>(grid,
+                                                                     threads)
+      .launch_test(init, sum, operand);
+  AtomicLauncher<atomic_fetch_sub_kernel<data_t, operand_t>, data_t>(grid,
+                                                                     threads)
+      .launch_test(sum, init, operand);
+  AtomicLauncher<atomic_fetch_add_kernel<data_t, operand_t, true>, data_t>(
+      grid, threads)
+      .launch_test(init, sum, operand);
+  AtomicLauncher<atomic_fetch_sub_kernel<data_t, operand_t, true>, data_t>(
+      grid, threads)
+      .launch_test(sum, init, operand);
+}
+
 int main() {
-  INSTANTIATE_ALL_TYPES(atomic_value_type_list, test_atomic_arith);
-  INSTANTIATE_ALL_TYPES(atomic_ptr_type_list, test_atomic_ptr_arith);
+  INSTANTIATE_ALL_TYPES(std::tuple<int>, test_atomic_arith);
+  // INSTANTIATE_ALL_TYPES(atomic_ptr_type_list, test_atomic_ptr_arith);
```

Comment:
I remember you mentioned that one of the changes broke `atomic_fetch_add` etc for pointer types, but I can't recall why? I think we decided we might need to remove the `arith_t`. Let's discuss this.